### PR TITLE
armandino: second submission

### PR DESCRIPTION
**Pull request by armandino**
#### Check List:
- [x] Tests pass (`./test.sh <username>`)
- [x] All formatting changes by the build are committed
- [x] Output matches that of calculate_average_baseline.sh
* Execution time:
```
real    0m44.224s
user    1m14.223s
sys     0m7.502s
```
* Execution time of reference implementation: ~5min

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_armandino.java [hunk 1]
```diff
@@ -18,173 +18,111 @@
...
-                }
-                else if (b == MINUS) {
-                    sample.sign = -1;
-                }
-                else {
-                    sample.pushMeasurement(b);
-                }
+            if (stats == null) {
+                stats = new Stats(new String(sample.keyBytes, 0, sample.keyLength, UTF_8));
+                map.put(sample.keyHash, stats);
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_armandino.java:108]: This seems prone to hash collisions? There could be multiple names with the same hash and results would be wrong. I suppose that actually was an issue with the initial implementation already.

**Review by gunnarmorling — CHANGES_REQUESTED** [review]

File: src/main/java/dev/morling/onebrc/CalculateAverage_armandino.java (hunk 1 above)
Comment (armandino) [reply src/main/java/dev/morling/onebrc/CalculateAverage_armandino.java:108]: @gunnarmorling thanks for the feedback. This should be equivalent to using `String` as a key if ingot this right. Maybe we need a test for collisions?

**Comment by gunnarmorling** [comment]
Can you please rebase this to current main to resolve the merge conflict? Thx.

**Comment by armandino** [comment]
Having problems with rebase. I'll close this and submit a new PR.

**Pull request closed without merging**